### PR TITLE
Allow callers of Targets() to exclude 'retry-never' peers

### PR DESCRIPTION
**Pull request by bboreham**
Required for https://github.com/weaveworks/weave/pull/2070

### CODE REVIEW

File: connection_maker.go [hunk 1]
```diff
@@ -135,14 +135,20 @@ func (cm *connectionMaker) ForgetConnections(peers []string) {
...
 // Note these are the same things that InitiateConnections and ForgetConnections talks about,
 // but a method to retrieve 'Connections' would obviously return the current connections.
-func (cm *connectionMaker) Targets() []string {
+func (cm *connectionMaker) Targets(activeOnly bool) []string {
 	resultChan := make(chan []string, 0)
 	cm.actionChan <- func() bool {
 		var slice []string
 		for peer := range cm.directPeers {
+			if activeOnly {
+				if target, found := cm.targets[peer]; found && target.tryAfter.IsZero() {
```
Comment (peterbourgon) [inline connection_maker.go]: s/found/ok

**Comment by peterbourgon** [comment]
After that little nit is picked, LGTM.

**Comment by bboreham** [comment]
Changed as requested.

**Comment by peterbourgon** [comment]
Merge at your discretion :)

**Comment by bboreham** [comment]
I do not have write access to this repository.

**Pull request merged**